garage/tf/algos/ddpg.py

```diff
@@ -240,89 +241,99 @@ def train(self, sess=None):
-            self.actor.global_vars, target_actor.global_vars, self.tau)
-        critic_init_ops, critic_update_ops = get_target_ops(
-            self.critic.global_vars, target_critic.global_vars, self.tau)
-        target_init_op = actor_init_ops + critic_init_ops
-        target_update_op = actor_update_ops + critic_update_ops
-
-        f_init_target = tensor_utils.compile_function(
-            inputs=[], outputs=target_init_op)
-        f_update_target = tensor_utils.compile_function(
-            inputs=[], outputs=target_update_op)
-
-        y = tf.placeholder(tf.float32, shape=(None, 1), name="input_y")
-        obs = tf.placeholder(
-            tf.float32,
-            shape=(None, self.observation_dim),
-            name="input_observation")
-        actions = tf.placeholder(
-            tf.float32, shape=(None, self.action_dim), name="input_action")
-
-        # Set up actor training function
-        next_action = self.actor.get_action_sym(obs, name="actor_action")
-        next_qval = self.critic.get_qval_sym(
-            obs, next_action, name="actor_qval")
-        action_loss = -tf.reduce_mean(next_qval)
-        if self.actor_weight_decay > 0.:
-            actor_reg = tc.layers.apply_regularization(
-                tc.layers.l2_regularizer(self.actor_weight_decay),
-                weights_list=self.actor.regularizable_vars)
-            action_loss += actor_reg
-
-        actor_train_op = self.actor_optimizer(
-            self.actor_lr, name="ActorOptimizer").minimize(
-                action_loss, var_list=self.actor.trainable_vars)
-
-        f_train_actor = tensor_utils.compile_function(
-            inputs=[obs], outputs=[actor_train_op, action_loss])
-
-        # Set up critic training function
-        qval = self.critic.get_qval_sym(obs, actions, name="q_value")
-        qval_loss = tf.reduce_mean(tf.squared_difference(y, qval))
-        if self.critic_weight_decay > 0.:
-            critic_reg = tc.layers.apply_regularization(
-                tc.layers.l2_regularizer(self.critic_weight_decay),
-                weights_list=self.critic.regularizable_vars)
-            qval_loss += critic_reg
-
-        critic_train_op = self.critic_optimizer(
-            self.critic_lr, name="CriticOptimizer").minimize(
-                qval_loss, var_list=self.critic.trainable_vars)
-
-        f_train_critic = tensor_utils.compile_function(
-            inputs=[y, obs, actions],
-            outputs=[critic_train_op, qval_loss, qval])
-
-        self.opt_info = dict(
-            f_train_actor=f_train_actor,
-            f_train_critic=f_train_critic,
-            f_init_target=f_init_target,
-            f_update_target=f_update_target,
-            replay_buffer=replay_buffer,
-            target_critic=target_critic,
-            target_actor=target_actor)
+        with tf.name_scope(self.name, "DDPG"):
+            """Set up the actor, critic and target network."""
```

Comment:
would a `tf.name_scope(self.name, "setup_networks")` be appropriate here? i.e. from lines 245 (here) to 258 (`target_critic._build_net(trainable=False)`)

Reply:
I could add a scope, but I think the name "networks" would be more appropiate, since the Actor, Critic and Targets that would be enclosed only contain the operations per layer in each network.
Do you agree?